concord-ui/src/components/statstable.js

```diff
@@ -0,0 +1,193 @@
+                                .map((v, i) => (
+                                    <DataTableRow key={i}>
+                                        <DataTableCell>
+                                            {data[i].userId}
+                                        </DataTableCell>
+                                        <DataTableCell alignEnd>
+                                            {data[i].totalVotes.toLocaleString()}
+                                        </DataTableCell>
+                                        <DataTableCell alignEnd>
+                                            {data[i].totalVotesWithConsensus.toLocaleString()}
+                                        </DataTableCell>
+                                        <DataTableCell alignEnd>
+                                            {data[i].completedVotes.toLocaleString()}
+                                        </DataTableCell>
+                                        <DataTableCell alignEnd>
+                                            {data[i].trashVotes.toLocaleString()}
+                                        </DataTableCell>
+                                        <DataTableCell alignEnd>
+                                            {this.toPercentage(data[i].completedVotes, data[i].totalVotesWithConsensus)}%
+                                        </DataTableCell>
+                                        <DataTableCell alignEnd>
+                                            {this.toPercentage(data[i].trashVotes, data[i].totalVotes)}%
+                                        </DataTableCell>
+                                        <DataTableCell alignEnd>
+                                            {this.toPercentage(data[i].completedVotesIgnoringTrash, data[i].totalVotesWithConsensusIgnoringTrash)}%
+                                        </DataTableCell>
+                                    </DataTableRow>
+                                ))}
+                        </DataTableBody>
+                    </DataTableContent>
+                </DataTable>
+            )
+        } else {
+            return (
+                <div><p>No stats to display</p></div>
+            )
+        }
+    }
+
+    /**
+     * Clears the column sort arrows
+     */
+    clearSorts() {
+        this.setState({
+            totalSortDir: null,
+            totalWithConsensusSortDir: null,
+            completedSortDir: null,
+            trashSortDir: null,
+            accuracyRateSortDir: null,
+            trashRateSortDir: null,
+            accuracyRateNoTrashSortDir: null,
+        })
+    }
+
+    /**
+     * Sort rows and update the data state, render() will then update the UI
+     */
+    sortRows = (sortDir, supplier) => {
+        const data = this.state.data;
+        data.sort((a, b) => sortDir * (supplier(a) - supplier(b)));
+        this.setState({data: data})
+    };
+
+    sortByTotal = (sortDir) => {
```

Comment:
You can generalise these functions using JS syntactic sugar.
e.g.
this.setState({ apple  })
does the same as this.setState({ apple: apple }).
Will just need to pass in a parameter.